_posts/2018-05-07-A-Machine-Learning-Example-For-Business.md

```diff
@@ -0,0 +1,271 @@
+of predicting by variables as it's very common
+scenario in business analytics. Also many of the
+concepts can be used even in image recognition once the
+image pixels has been converted to tabular data.
+
+Let's imagine that we are running an online shop that
+orders nutrition supplements from a wholesaler and sells the products to consumers.
+
+![Machine learning on supply chain management](/img/a-machine-learning-example-for-business/machine-learning-supply-chain.png)
+
+The online store wants to <b>maximize those wholesale orders that will arrive on time</b>. This increases predictability and that way the satisfaction in end users and sales as well.
+
+## Sample data
+In the sample data a record corresponds to an order made for a wholesaler in the past. The first column is the week when the order has been made (`week`). Then there are the day of week (`order_day`), order batch size (`order_size`), sub contractor delivering the order (`sub_contractor`) and the info whether the delivery was late or not (`is_late`).
+
+![Sample data for machine learning](/img/a-machine-learning-example-for-business/sample-data.png)
+
+Usually collecting and cleaning the data is by far the most time consuming part. That has been already done in this example, so we can move to analysis. For example, maybe in the original data the size of order was the number of items, but for this case the variable has expressed in S (small), M (medium) or L (large).
+
+There are dozens of predictive machine learning algorithms that work in a different way, but a typical situation is to create a table like this, where the last column is the variable that you want to predict. For historical data the predicted variable is known already. All or part of the other variables are predictor variables.
+
+![Result variable in sample data](/img/a-machine-learning-example-for-business/sample-data-result-variable.png)
+
+## Predicting from reports is not systematic
+In traditional reporting you could analyze the rate of late arriving orders from previously introduced 20 rows of data by doing basic data aggregation and see how many percent of orders have arrived late on average for each variable combination.
+
+![Summary report](/img/a-machine-learning-example-for-business/summary-report.png)
+
+Using reports to forecast future is problematic, however. If there's only one record for a combination such as `Mon`, `L` and `Express` in row 1, the probability is always 0 or 1, which can't be true. The summary report also does not handle individual variables (columns) as independent entities, and that's why there is no way of evaluating probability for previously non existing combinations such as `Tue`, `M` and `Go`.
+
+And the most important part: There's no way of evaluating how these insights fit into new wholesale orders.
+
+## Concept of training and testing in machine learning
+The secret of predictive machine learning is to split the process in two phases:
```

Comment:
A bit nitpicking here, but I would speak of "supervised machine learning" instead of predictive machine learning here.

Reply:
It would be great to involve the differences between supervised and unsupervised methods, but as this is targeted to people in between business and tech, I will leave this part to current state.